File: src/01-easy/moves-in-maze/solution.cs
using System;
using System.Collections.Generic;
using System.Text;
class Solution {
    static void Main(){
        string C="0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        var p=Console.ReadLine().Split(' ');int w=int.Parse(p[0]),h=int.Parse(p[1]);
        var g=new string[h];for(int i=0;i<h;i++) g[i]=Console.ReadLine();
        var d=new int[h,w];for(int r=0;r<h;r++)for(int c=0;c<w;c++)d[r,c]=-1;
        int sr=0,sc=0;
        for(int r=0;r<h;r++)for(int c=0;c<w;c++)if(g[r][c]=='S'){sr=r;sc=c;}
        d[sr,sc]=0;
        var q=new Queue<(int,int)>();q.Enqueue((sr,sc));
        int[] dr={0,0,1,-1},dc={1,-1,0,0};
        while(q.Count>0){var(r,c)=q.Dequeue();for(int i=0;i<4;i++){int nr=(r+dr[i]+h)%h,nc=(c+dc[i]+w)%w;if(g[nr][nc]!='#'&&d[nr,nc]==-1){d[nr,nc]=d[r,c]+1;q.Enqueue((nr,nc));}}}
        var sb=new StringBuilder();
        for(int r=0;r<h;r++){for(int c=0;c<w;c++)sb.Append(g[r][c]=='#'?'#':d[r,c]==-1?'.':C[d[r,c]]);sb.AppendLine();}
        Console.Write(sb);
    }
}

File: src/01-easy/moves-in-maze/solution.vb
Module Solution
    Sub Main()
        Dim chars As String = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ"
        Dim p = Console.ReadLine().Split(" "c)
        Dim w = CInt(p(0)), h = CInt(p(1))
        Dim g(h-1) As String
        Dim d(h-1, w-1) As Integer
        For i = 0 To h-1
            g(i) = Console.ReadLine()
            For j = 0 To w-1
                d(i,j) = -1
            Next
        Next
        Dim sr = 0, sc = 0
        For r = 0 To h-1
            For c2 = 0 To w-1
                If g(r)(c2) = "S"c Then
                    sr = r
                    sc = c2
                End If
            Next
        Next
        d(sr, sc) = 0
        Dim q As New Queue(Of Tuple(Of Integer, Integer))
        q.Enqueue(Tuple.Create(sr, sc))
        Dim dr = {0, 0, 1, -1}, dc = {1, -1, 0, 0}
        While q.Count > 0
            Dim pt = q.Dequeue()
            Dim cr = pt.Item1, cc = pt.Item2
            For i = 0 To 3
                Dim nr = (cr+dr(i)+h) Mod h, nc = (cc+dc(i)+w) Mod w
                If g(nr)(nc) <> "#"c AndAlso d(nr, nc) = -1 Then
                    d(nr, nc) = d(cr, cc)+1
                    q.Enqueue(Tuple.Create(nr, nc))
                End If
            Next
        End While
        For r = 0 To h-1
            Dim sb As New System.Text.StringBuilder()
            For c2 = 0 To w-1
                If g(r)(c2) = "#"c Then
                    sb.Append("#"c)
                ElseIf d(r,c2) = -1 Then
                    sb.Append("."c)
                Else
                    sb.Append(chars(d(r,c2)))
                End If
            Next
            Console.WriteLine(sb.ToString())
        Next
    End Sub
End Module